code/controllers/subsystem/atoms.dm

```diff
@@ -52,6 +52,7 @@ SUBSYSTEM_DEF(atoms)
 			if(QDELETED(A))
 				continue
 			A.LateInitialize()
+			CHECK_TICK
```

Comment:
I asked myself a billion times why this wasn't already the case. Another source of tick overrun was not checking between these calls.